lib/NodeStuffPlugin.js

```diff
@@ -80,13 +80,15 @@ NodeStuffPlugin.prototype.apply = function(compiler) {
 		this.state.module.warnings.push(new UnsupportedFeatureWarning(this.state.module, "require.extensions is not supported by webpack. Use a loader instead."));
 		return true;
 	});
-	compiler.parser.plugin("expression module.exports", ignore);
-	compiler.parser.plugin("expression module.loaded", ignore);
-	compiler.parser.plugin("expression module.id", ignore);
-	compiler.parser.plugin("evaluate Identifier module.hot", function(expr) {
-		return new BasicEvaluatedExpression().setBoolean(false).setRange(expr.range);
-	});
-	compiler.parser.plugin("expression module", function() {
-		return ModuleParserHelpers.addParsedVariable(this, "module", "require(" + JSON.stringify(path.join(__dirname, "..", "buildin", "module.js")) + ")(module)");
-	});
+	if(this.options.module) {
+		compiler.parser.plugin("expression module.exports", ignore);
+		compiler.parser.plugin("expression module.loaded", ignore);
+		compiler.parser.plugin("expression module.id", ignore);
+		compiler.parser.plugin("evaluate Identifier module.hot", function(expr) {
+			return new BasicEvaluatedExpression().setBoolean(false).setRange(expr.range);
+		});
```

Comment:
The first 3 are part of CommonJs. The 4th is part of HMR. You propably only want to disable this last one.